File: internal/parser/testdata/sample.go
package sample

import (
	"fmt"
	"net/http"
)

const MaxRetries = 3

type Config struct {
	Host string
	Port int
}

type Handler interface {
	ServeHTTP(w http.ResponseWriter, r *http.Request)
}

var defaultConfig = Config{Host: "localhost", Port: 8080}

func NewConfig() *Config {
	return &defaultConfig
}

func (c *Config) Run() error {
	addr := fmt.Sprintf("%s:%d", c.Host, c.Port)
	return http.ListenAndServe(addr, nil)
}

File: internal/parser/testdata/sample.php
<?php

use App\Models\User;

const MAX_RETRIES = 3;

function create_config(string $host, int $port): array {
    return ['host' => $host, 'port' => $port];
}

interface Handler {
    public function handle($request);
}

trait Loggable {
    public function log(string $message): void {
        echo $message;
    }
}

class Config implements Handler {
    use Loggable;

    private string $host;
    private int $port;

    public function __construct(string $host, int $port) {
        $this->host = $host;
        $this->port = $port;
    }

    public function address(): string {
        return $this->host . ':' . $this->port;
    }

    public function handle($request) {
        $this->log('handling');
    }
}
